refactor(mobile): bring back backup selected local assets to timeline

shenlong-tanwen opened the pull request:
Follows #7000

#### Changes made:

- Local albums will not be rendered in the main timeline after 7000. This PR brings it back. After this change:
   - Assets that are a part of backup will be displayed in the main timeline
   - The remaining local assets can be accessed in Library -> Albums on Device even if they are not selected for backup. This would allow users to select and manually upload assets instead of relying on auto backup
- Moves the filtering asset to backup logic during album selection. (This would help us filter assets to be backed up during the actual backup process much easily. This will be added in a follow up PR)

shenlong-tanwen committed:
feat(mobile): select which local assets to display in timeline

shenlong-tanwen committed:
remove album selection chips

fyfrey commented on mobile/lib/shared/models/asset.dart:
```diff
@@ -35,7 +37,8 @@ class Asset {
         stackParentId = remote.stackParentId,
-        stackCount = remote.stackCount;
+        stackCount = remote.stackCount,
+        selectedForBackup = BackupSelection.none;
```
I'd really like to avoid adding this field to the asset class. Can we store this elsewhere?
E.g. a list of asset IDs with each selection state?

fyfrey commented on mobile/lib/shared/models/asset.dart:
```diff
@@ -185,6 +190,12 @@ class Asset {
+  BackupSelection selectedForBackup;
+
+  @Backlink(to: Album.assetsLinkId)
+  final IsarLinks<LocalAlbum> localAlbums = IsarLinks<LocalAlbum>();
```
I'd like to avoid this as well as we can't move assets across isolates when they contain links

shenlong-tanwen committed:
refactor: move backup selection to device asset

shenlong-tanwen commented:
@fyfrey Did the following changes to address all the comments:

- Moved backupSelection state to `DeviceAsset`
- Unified `AndroidDeviceAsset` and `IOSDeviceAsset` back into `DeviceAsset`. Changed the Isar ID to AutoIncrement. We might need to migrate the DB and recalculate all hash on device once. Will add this change if we are okay with proceeding with this change
- Removed the Isar links from asset collection and manually queries the localAlbums db when required

fyfrey commented on mobile/lib/shared/providers/asset.provider.dart:
```diff
@@ -377,16 +384,27 @@ QueryBuilder<Asset, Asset, QAfterSortBy>? getRemoteAssetQuery(WidgetRef ref) {
       .isArchivedEqualTo(false)
       .isTrashedEqualTo(false)
       .stackParentIdIsNull()
+      .group(
```
What's the purpose of group operation here?

shenlong-tanwen replied:
It is to make the timeline related filter for local assets more obvious.

fyfrey reviewed (approved):
Thanks for making these changes. Code looks good, didn't run it, though.
